common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -405,19 +469,35 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
+                    result = DeadlineRunResult.TIMED_OUT;
```

Comment:
> I can imagine a caller which is just interested in the status of executing these tasks as opposed to finding the times, in which case using a callback for getting the third status is awkward.

That is what `runAllTasks()` is for no? It seems like if timing is involved then the execution time should not be ignored. For example I'm not sure why we don't preserve the execution time when running the `tailTasks` (the first bullet above).

> This can probably be questioned BUT the case in point being that the callback is optional

Can you explain why it is optional?